File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        if (prices.size() < 2) {
            return 0;
        }

        int lowest = prices[0];
        int max_profit=0;
        for (int price : prices) {

            if (price < lowest) { 
                lowest = price;
                continue;
            } else {
                max_profit = max(max_profit, price - lowest);
            }
        }

        return max_profit;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sett = set()

        n = len(s)
        l, r = 0, 0

        out = 0
        while l < n and r < n:
            if s[r] not in sett:
                sett.add(s[r])
                r += 1
            else:
                while s[l] != s[r] and l < r:
                    sett.remove(s[l])
                    l += 1
                sett.remove(s[l]) 
                l += 1
            out = max(out, r - l)
        return out